### PR TITLE
/rank format targeting

### PR DESCRIPTION
**Pull request by YaYana**
Seeing how nobody worked on #251 anymore, I've implemented what was asked.
It was a bit tricky because of the HTML sanitizing though, making this not the most elegant solution, I guess...

### CODE REVIEW

File: js/client-chat.js [hunk 1]
```diff
@@ -641,8 +658,31 @@
...
 						buffer += '</table></div>';
 						self.add('|raw|'+buffer);
+
+						if (hiddenFormats.length) {
+							self.$chatFrame.find('button.show-other-formats:last').click(function (e) {
+								var autoscroll = (self.$chatFrame.scrollTop() + 60 >= self.$chat.height() - self.$chatFrame.height());
+
+								var $target = $(e.currentTarget);
+								var $table = $target.closest('table');
+								$table.find('tr.hidden').css('display', 'table-row');
```
Comment (Zarel) [inline js/client-chat.js]: lol, I was about to say "not IE6/IE7-compatible" before remembering we don't have to worry about supporting them anymore.

Comment (Zarel) [inline js/client-chat.js]: Still, though, wouldn't `.show()` be easier?

File: js/client-chat.js [hunk 2]
```diff
@@ -641,8 +658,31 @@
...
+								buffer += '<tr class="no-matches"><td colspan="8"><em>This user has not played any ladder games that match the format targeting.</em></td></tr>';
+							}
+
+							buffer += '<tr><td colspan="8"><button class="show-other-formats">' + hiddenFormats.slice(0, 3).join(', ') + (hiddenFormats.length > 3 ? ' and ' + (hiddenFormats.length - 3) + ' other formats' : '') + ' not shown</button></td></tr>';
+						}
 						buffer += '</table></div>';
 						self.add('|raw|'+buffer);
+
+						if (hiddenFormats.length) {
+							self.$chatFrame.find('button.show-other-formats:last').click(function (e) {
```
Comment (Zarel) [inline js/client-chat.js]: This appears to be the wrong place to attach an event listener.

Comment (Zarel) [inline js/client-chat.js]: You don't even need another event listener; presumably our `button name` system should be enough.

**Comment by YaYana** [comment]
I was unaware of the `button name` system. It's really nifty.
Anyhow, I've made the changes you requested.

File: js/client-chat.js [hunk 3]
```diff
@@ -607,18 +607,35 @@
...
-							for (var i=0; i<data.length; i++) {
+							var hiddenFormats = [];
+							var formatLength = data.length;
+							for (var i=0; i<formatLength; i++) {
 								var row = data[i];
-								buffer += '<tr><td>'+row.formatid+'</td><td><strong>'+Math.round(row.acre)+'</strong></td><td>'+Math.round(row.gxe,1)+'</td><td>';
+								if (!formatTargeting || formats[row.formatid]) {
+									buffer += '<tr>';
+								} else {
+									buffer += '<tr class="hidden">';
```
Comment (Zarel) [inline js/client-chat.js]: You probably shouldn't use the HTML sanitizer, then... There are a few workarounds.

If you can't figure out any other workaround, I'd accept replacing `.ladder .hidden {` with `.hidden {` in the CSS files.

Comment (YaYana) [inline js/client-chat.js]: I couldn't figure out a way to make it not use the HTML sanitizer without changing loads of functions or practically copy-pasting whole functions, so I opted for your alternative.

**Pull request merged**